### ISSUE TITLE
dom0 kernel 5.10.11-1.fc25 from current-testing -> libxenxl errors (revert to 5.10.8-1 resolves)

### ISSUE THREAD

**Issue by brendanhoar** [QubesOS/qubes-issues#6377]
**Qubes OS version**
R4.0 current-testing

**Affected component(s) or functionality**
domU startup

**Brief summary**
all VMs (except dom0) fail to start up.

**How Reproducible**
Every reboot

**To Reproduce**
install dom0 kernel 5.10.11.1.fc25 via current-testing repo

1. Do above.
2. Attempt to launch a VM.

**Expected behavior**
VM launches

**Actual behavior**
The following errors in [/var/log/libvirt/libxl/libxl-driver.log]
```
2021-01-30 21:24:25.125+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to add device with path /local/domain/0/backend/vbd/5/51712
2021-01-30 21:24:25.167+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to add device with path /local/domain/0/backend/vbd/5/51728
2021-01-30 21:24:25.217+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to add device with path /local/domain/0/backend/vbd/5/51744
2021-01-30 21:24:25.339+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to add device with path /local/domain/0/backend/vbd/5/51760
2021-01-30 21:24:25.339+0000: libxl: libxl_create.c:1320:domcreate_launch_dm: unable to add disk devices
2021-01-30 21:24:35.387+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to remove device with path /local/domain/0/backend/vbd/5/51712
2021-01-30 21:24:35.430+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to remove device with path /local/domain/0/backend/vbd/5/51728
2021-01-30 21:24:35.454+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to remove device with path /local/domain/0/backend/vbd/5/51744
2021-01-30 21:24:35.477+0000: libxl: libxl_device.c:1099:device_backend_callback: unable to remove device with path /local/domain/0/backend/vbd/5/51760
2021-01-30 21:24:35.505+0000: libxl: libxl.c:1669:devices_destroy_cb: libxl__devices_destroy failed for 5
```

**Additional context**
Rebooting and opting for an older kernel (e.g. 5.10.8-1) resolves the issue

**Comment by marmarek** [comment]
Known problem: https://lore.kernel.org/xen-devel/983b87d6-edb8-21ea-7d6f-f653f5c0d048@oracle.com/T/#t
The fix missed 5.10.12, but should be in 5.10.13.

**Comment by DemiMarie** [comment]
Can we backport the patch to our own 5.10.1{1,2} builds?

**Comment by FabrizioRomanoGenovese** [comment]
I all, I have exactly the same problem, but opting for another kernel doesn't help me: The other two available kernels for me are
`5.4.90-1.qubes.x86_64` and `5.4.91-1.fc25.qubes.x86_64`. I tried to boot switching to either of the two by editing `xen.cfg` in `/boot/efi/EFI/qubes/` using a live distro. What happens is that I get past the LUCKS authentication screen, but I get stuck at the second login screen (when it asks for user and pass): This is black screen, I can see only the mouse, and is so slow to be unusable.

I understand the current kernel has a known back, but how can I get out of this situation? The other kernels do not seem to work at all, and not being able to launch any vm I don't know how will I be able to install the new kernel on my system once updates become available. Any suggestion is much appreciated!

**Comment by marmarek** [comment]
I assume 5.10.11 is not the first version of kernel-latest package you installed (otherwise you wouldn't have working system before based on your description). The package manager keeps up to three older kernels for each flavor (latest, default etc), which means you should have previous kernel-latest (5.10.8?) still somewhere installed. If you can't see it in `/boot/efi/EFI/qubes`, perhaps it didn't fit there, but it should still be in /boot. If that's the case, you'll need to copy appropriate vmlinuz and initramfs file manually and add a section to xen.cfg.

I the meantime 5.10.13 should be available (it's already building, but it takes few hours).

**Comment by FabrizioRomanoGenovese** [comment]
Unfortunately that's not the case T_T The only things that weren't working in my system with 5.4 were wifi and suspend. Since my wifi drivers should theoretically come with kernel 5.10, I decided to give it a shot installing it in sys-net. When the problem persisted, I tried switching to 5.10 also in dom0. So, unfortunately, I just have the three kernels I listed above...

**Comment by FabrizioRomanoGenovese** [comment]
...Whohoo! Solved! So, what I did is the following:

I got into my machine with kernel 5.4. I found out that giving Cntrl+Alt+F2 gave me a shell, which greeted me by saying that X was hopeless. Then I noticed that in kernel 5.4 I was able to launch VMs like vault, that weren't connected to any PCI device. Still no hope for sys-net unfortunately. I fiddled with the [PCI Throubleshooting page](https://www.qubes-os.org/doc/pci-troubleshooting/) and managed to start sys-net, and got internet back. From here, I just waited a few hours and downloaded the new kernel latest, 5.10.13-1, that as predicted solved the problem. :)
